osquery/config/config.cpp

```diff
@@ -137,17 +156,19 @@ std::map<std::string, std::vector<std::string> >& Config::getWatchedFiles() {
   return getInstance().cfg_.eventFiles;
 }
 
+pt::ptree& Config::getEntireConfiguration() {
+  boost::shared_lock<boost::shared_mutex> lock(rw_lock);
```

Comment:
This lock will expired when it goes out of scope, but the reference to the property tree will now exist both within the config (when you lock before access) AND with whomever called `getEntireConfiguration`. Return a new instance of the tree, or some RAII-locking object.